### ISSUE TITLE
[Improvement] Improve GravitinoCatalogManager close method

### ISSUE THREAD

**Issue by justinmclean** [datastrato/gravitino#2911]
### What would you like to be improved?

GravitinoCatalogManager's close method mixes static and volatile variables in a nonstatic method and that may cause issues. Correctly updating a static field from a non-static method can be tricky to get right and could easily lead to bugs if there are multiple threads.

### How should we improve?

See if this can be improved to be more thread safe.

**Comment by xiaozcy** [comment]
I created a PR(#3096) for this, could you take a look at it?

**Comment by FANNG1** [comment]
it's not supposed to work in the multi-thread environment, so I prefer no to change it.

**Comment by justinmclean** [comment]
Is the fact that it won't work in a multi-thread environment documented anywhere? That seems like an omission that a user would want to know about.

**Comment by FANNG1** [comment]
There is no document for this, maybe we could add some comment to the class file, WDYT?

**Comment by yuqi1129** [comment]
> See if this can be improved to be more thread safe.

Is there a need to support multi-threading? It currently does not support multi-threading.

**Comment by justinmclean** [comment]
Perhaps I'm misisng something, but I assume users expect this to support muti-threading.

**Comment by shaofengshi** [comment]
I agree with Justin, we couldn't stop user/client (especially it is a flink connector) from using it in multi-threading, so the code should be robust, especially that is too complex to achieve.
